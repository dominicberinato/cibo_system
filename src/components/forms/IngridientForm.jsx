import React, {Component} from 'react'
import {Field, reduxForm} from 'redux-form'
import {required} from 'validation'
import {renderField} from 'renderField'

export class IngridientForm extends Component {
  render() {
    var {handleSubmit} = this.props;
    return(
      <div>
        <div className="text-center form-label">
          <p>Add Ingridient</p>
        </div>
        <form ref="form" onSubmit={handleSubmit}>
        <div>
          <label>Category</label>
          <Field name="ingredientCategory" type="text" component={renderField} validate={[required]}/>
        </div>
        <div>
          <label>Description</label>
          <Field name="ingredientDescription" type="text" rows="4" component={renderField} validate={[required]}/>
        </div>
        <div>
          <label>Brand</label>
          <Field name="ingredientBrand" type="text" component={renderField} validate={[required]}/>
        </div>
        <div>
          <label>Pack Size</label>
          <Field name="ingredientPSize" type="text" component={renderField} validate={[required]}/>
        </div>
        <div>
          <label>Pack Cost</label>
          <Field name="ingredientPackCost" type="text" component={renderField} validate={[required]}/>
        </div>
        <div>
          <label>Supplier</label>
          <Field name="ingredientSupplier" type="text" component={renderField} validate={[required]}/>
        </div>
        <div>
          <label>Estimate Life Span</label>
          <Field name="ingredientElife" type="text" component={renderField} validate={[required]}/>
        </div>
        <div className="text-center">
          <input type="submit" value="Add Ingridient" className="button"/>
        </div>
        </form>
      </div>
    )
  }
};

IngridientForm = reduxForm({
  form:'ingridient'
})(IngridientForm)

export default IngridientForm;
