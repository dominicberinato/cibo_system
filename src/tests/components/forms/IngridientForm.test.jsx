import expect from 'expect'
import {IngridientForm} from 'IngridientForm'
import {mount} from 'enzyme'
import React from 'react'
import sinon from 'sinon'

describe.only('<IngridientForm/>', () => {
  it('should exist', () => {
    expect(IngridientForm).toExist();
  });

})
