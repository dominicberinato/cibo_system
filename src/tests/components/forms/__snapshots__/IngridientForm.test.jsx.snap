// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IngridientForm/> <IngridientForm/> to render form correctly 1`] = `
<div>
  <p>
    Add Ingredient
  </p>
  <form>
    <div>
      <label>
        Category
      </label>
      <Field
        component={[Function]}
        name="ingredientCategory"
        type="text"
        validate={
          Array [
            [Function],
          ]
        }
      />
    </div>
    <div>
      <label>
        Description
      </label>
      <Field
        component={[Function]}
        multiLine={true}
        name="ingredientDescription"
        rows={4}
        type="text"
        validate={
          Array [
            [Function],
          ]
        }
      />
    </div>
    <div>
      <label>
        Brand
      </label>
      <Field
        component={[Function]}
        name="ingredientBrand"
        type="text"
        validate={
          Array [
            [Function],
          ]
        }
      />
    </div>
    <div>
      <label>
        Pack Units
      </label>
      <Field
        component={[Function]}
        name="ingredientPUnits"
        placeholder="kgs, g, lts, mls, spoons"
        type="text"
        validate={
          Array [
            [Function],
          ]
        }
      />
    </div>
    <div>
      <label>
        Pack Size
      </label>
      <Field
        component={[Function]}
        name="ingredientPSize"
        type="text"
        validate={
          Array [
            [Function],
          ]
        }
      />
    </div>
    <div>
      <label>
        Pack Cost
      </label>
      <Field
        component={[Function]}
        name="ingredientPackCost"
        type="text"
        validate={
          Array [
            [Function],
          ]
        }
      />
    </div>
    <div>
      <label>
        Supplier
      </label>
      <Field
        component={[Function]}
        name="ingredientSupplier"
        type="text"
        validate={
          Array [
            [Function],
          ]
        }
      />
    </div>
    <div>
      <label>
        Estimate Life Span
      </label>
      <Field
        component={[Function]}
        format={[Function]}
        name="ingredientElife"
        type="text"
        validate={
          Array [
            [Function],
          ]
        }
      />
    </div>
    <div>
      <label>
        Allergens
      </label>
      <Field
        component={[Function]}
        name="ingredientAllergen"
        placeholder="Milk, Sugar,Gluten"
        type="text"
      />
    </div>
    <div
      className="text-center"
    >
      <input
        className="button"
        type="submit"
        value="Add Ingridient"
      />
    </div>
  </form>
</div>
`;
